File: base-repository/legacy-app/src/main/java/alytvyniuk/com/base/LegacyApplication.java
package alytvyniuk.com.base;

import android.util.Log;

import alytvyniuk.com.base.di.LegacyDiComponent;
import dagger.android.DaggerApplication;

public abstract class LegacyApplication extends DaggerApplication {

    private static final String TAG = "LegacyApplication";

    private static LegacyApplication instance;

    public static LegacyApplication getInstance() {
        return instance;
    }

    public static LegacyDiComponent getRingApplicationComponent() {
        return (LegacyDiComponent) instance.applicationInjector();
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        Log.d(TAG, "App onCreate, initialize required libraries");
    }
}

File: base-repository/legacy-app/src/main/java/alytvyniuk/com/base/LegacyApplication.kt
package alytvyniuk.com.base


import android.util.Log
import dagger.android.DaggerApplication

private const val TAG = "LegacyApplication"

abstract class LegacyApplication: DaggerApplication() {

    override fun onCreate() {
        super.onCreate()
        Log.d(TAG, "App onCreate, initialize required libraries")
    }
}